Terminal.Gui/ConsoleDrivers/WindowsDriver.cs

```diff
@@ -1045,33 +1083,40 @@ KeyCode MapKey (WindowsConsole.ConsoleKeyInfoEx keyInfoEx)
 				return ConsoleKeyMapping.MapToKeyCodeModifiers (keyInfo.Modifiers, (KeyCode)((uint)KeyCode.D0 + delta));
 			}
 			if ((keyInfo.Modifiers & (ConsoleModifiers.Alt | ConsoleModifiers.Control)) != 0) {
-				if (keyInfo.KeyChar == 0 || keyInfo.KeyChar == 30 || keyInfo.KeyChar == ((uint)KeyCode.D0 + delta)) {
+				// TODO: figure out why VK.ACCEPT (IME ACCEPT) is special cased below. 
+				if (keyInfo.KeyChar == 0 || keyInfo.KeyChar == (uint)VK.ACCEPT || keyInfo.KeyChar == ((uint)KeyCode.D0 + delta)) {
 					return ConsoleKeyMapping.MapToKeyCodeModifiers (keyInfo.Modifiers, (KeyCode)((uint)KeyCode.D0 + delta));
 				}
 			}
-			return ConsoleKeyMapping.MapToKeyCodeModifiers (keyInfo.Modifiers, (KeyCode)((uint)keyInfo.KeyChar));
+			return (KeyCode)((uint)keyInfo.KeyChar);
```

Comment:
Here you aren't been coherent again with the range A..Z when only the shift key is pressed with a printable key, by adding and removing the shift when send to Key event. I know you initially said that the `IsKeyCodeAtoZ` property was only for unit test only, but it's public and for me it's a rule that also should be used on others situations. I think we have a great disagreement on this. Why not follow the Windows rule in these situations as `ConsoleKeyInfo` does?
 
![image](https://github.com/BDisp/Terminal.Gui/assets/13117724/a6a66334-e7a0-493d-ad87-631882336208)

Reply:
Please give me yes or no answers to each of these questions:

- Do you agree the persistence representation of the example here (D1 with the shift key pressed) is "Shift+1"?

- Do you agree no Terminal.Gui app should ever threat "D1 with the shift key pressed" as "Shift+!"?

- Do you agree "Shift+!" is nonsensical?

- Do you agree `Key('!')` is NOT the same as `Key.D1.WithShift`?

If you answer yes to all of these questions, we don't have a deep disagreement. I believe we should honor the above by stripping shift off at the driver level. You think we should leave shift encoded in `Key.KeyCode` and have `ToString` (and `TryParse`) ensure it is ignored. I do not feel super strongly either way, but I do feel very strongly that the answer to each of the 4 questions above is YES.

If you answer NO to any of the above questions, we need to continue to debate this.

Reply:
> Please give me yes or no answers to each of these questions:
> 
> * Do you agree the persistence representation of the example here (D1 with the shift key pressed) is "Shift+1"?

Yes.

> * Do you agree no Terminal.Gui app should ever threat "D1 with the shift key pressed" as "Shift+!"?

Yes.

> * Do you agree "Shift+!" is nonsensical?

Yes.

> * Do you agree `Key('!')` is NOT the same as `Key.D1.WithShift`?
> 
Yes.

> If you answer yes to all of these questions, we don't have a deep disagreement. I believe we should honor the above by stripping shift off at the driver level. You think we should leave shift encoded in `Key.KeyCode` and have `ToString` (and `TryParse`) ensure it is ignored. I do not feel super strongly either way, but I do feel very strongly that the answer to each of the 4 questions above is YES.
> 
You are absolutely right.

> If you answer NO to any of the above questions, we need to continue to debate this.

Not much, only maybe on some things.

Reply:
Glad we agree :-).

Now, for the part we don't agree on...

Using "Shift+7" as an example (better example than "Shift+1" because "Shift+7" is different on ENG ('&') and POR ('/')):

The `keyInfo` will be:

```
Modifiers: Shift
Key: D7
KeyChar: 38 '&'
```

If we were to preserve the fact the shift key was down within `Key`, how would you propose we encode that in `Key.KeyCode`?

a) `Key.KeyCode = keyInfo.Key | KeyCode.ShiftMask`

b) `Key.KeyCode = keyInfo.KeyChar | KeyCode.ShiftMask`

If a), then the fact the keyboard layout specifies the char is '&' is lost.

If b), then the value in `Key.KeyCode` is `& | KeyCode.ShiftMask`. There is no way to serialize this as "Shift+7" because the virtual key code was lost.

We could solve this by adding another member to `Key` so it holds BOTH virtual key code and keychar. Or we could add more complex encoding to the `uint` that is `Key.KeyCode`. 

Can you think of a real-world use case (NOT the VK simulator!) where an app needed to know '&' (or '/') was generated by the user pressing `Shift+7`?

Reply:
This isn't not so easy to decide, but we really have to decide some solution. It's like we are re-event what already exist.

> If a), then the fact the keyboard layout specifies the char is '&' is lost.
> 
If we use this we have it as encoded and we have to decode to send the char into the event. I think this is onerous because we don't need to encode, base on the keyboard and decode to send to and event, other than Packet key.

> If b), then the value in `Key.KeyCode` is `& | KeyCode.ShiftMask`. There is no way to serialize this as "Shift+7" because the virtual key code was lost.
> 
This will be the more faster and used in the events.

> We could solve this by adding another member to `Key` so it holds BOTH virtual key code and keychar. Or we could add more complex encoding to the `uint` that is `Key.KeyCode`.
> 
It's a good solution because it belong to the `Key` class and have more than enough length to accommodate the key code and the keychar.

> Can you think of a real-world use case (NOT the VK simulator!) where an app needed to know '&' (or '/') was generated by the user pressing `Shift+7`?

Not really. I would like to know because for certain I would like to learning more about how to do this. A suggestion is a database based on the `GetKeyboardLayoutName` with the right char that belong to a certain layout, but it's very onerous to make it. If there is any function that already do that will be good.

Reply:
> > Can you think of a real-world use case (NOT the VK simulator!) where an app needed to know '&' (or '/') was generated by the user pressing `Shift+7`?
> 
> Not really.

Until there is a real-world use-case, I want to go with my current solution. My solution solves all known current solutions and is relatively elegant and simple. 

We can re-visit this later if some user comes along with a real-world need.